src/dragonball/src/vm/mod.rs

```diff
@@ -350,6 +350,15 @@ impl Vm {
         instance_state == InstanceState::Running
     }
 
+    /// Return true if TDX is enabled
+    pub fn is_tdx_enable(&self) -> bool {
+        let shared_info = self
+            .shared_info()
+            .read()
+            .expect("failed to get instance state, because shared info is poisoned lock");
```

Comment:
Checking for the TDX or TDX Seam module wouldn't be a better way to do this?

Reply:
Hi, here we are not checking whether the hardware has TDX cabability.
Instead, we are just checking the VM type is TDX VM or normal VM (The information should be passed by the runtime)
I will write more accurate comments and commit message and please tell me if that makes sense.

Reply:
Optimized. Please take a look.